kv/etcd/store.go

```diff
@@ -381,19 +422,19 @@ func (c *client) update(key string) error {
 	}
 
 	curValue := w.Get()
-	if curValue == nil && newValue == nil {
-		// nothing to update
+
+	// They have the same values, or both are nil. There is nothing to do.
+	if curValue == nv {
```

Comment:
Probably not, I just figured it should get handled the same way if it ever is. I'll change it back to check for nils